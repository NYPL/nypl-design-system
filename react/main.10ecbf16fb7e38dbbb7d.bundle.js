(window.webpackJsonp=window.webpackJsonp||[]).push([[0],{1292:function(n,o,c){"use strict";c.r(o);c(1293),c(1323),c(1335),c(1442),c(1485)},1448:function(n,o){},1449:function(n,o){},1450:function(n,o){},1451:function(n,o){},501:function(n,o,c){c(502),c(645),n.exports=c(1292)},564:function(n,o){}},[[501,1,2]]]);