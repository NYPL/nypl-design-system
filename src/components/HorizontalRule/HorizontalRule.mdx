import { ArgTypes, Canvas, Description, Meta } from "@storybook/blocks";

import * as HorizontalRuleStories from "./HorizontalRule.stories";
import Link from "../Link/Link";
import { changelogData } from "./horizontalRuleChangelogData";
import ComponentChangelogTable from "../../utils/ComponentChangelogTable";

<Meta of={HorizontalRuleStories} />

# HorizontalRule

| Component Version | DS Version   |
| ----------------- | ------------ |
| Added             | `0.23.0`     |
| Latest            | `Prerelease` |

## Table of Contents

- {<Link href="#overview" target="_self">Overview</Link>}
- {<Link href="#component-props" target="_self">Component Props</Link>}
- {<Link href="#accessibility" target="_self">Accessibility</Link>}
- {<Link href="#changelog" target="_self">Changelog</Link>}

## Overview

<Description of={HorizontalRuleStories} />

The `HorizontalRule` component renders a basic `<hr>` element. The `height`,
`width`, and `align` props can be customized. For the `height` prop, use a whole
number, a `px` value, a `em` value, or a `rem` value. Otherwise, the default
value of "2px" will be used.

## Component Props

<Canvas of={HorizontalRuleStories.WithControls} />

<ArgTypes of={HorizontalRuleStories.WithControls} />

## Accessibility

The `HorizontalRule` component is not just a visual thematic break between
content or interfaces, it is also picked up by screenreaders. Internally, the
`HorizontalRule` component uses the `<hr>` element which has a `role` of
`"separator"`.

Resources:

- [MDN hr: The Thematic Break (Horizontal Rule) element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/hr)
- [Not Your Typical Horizontal Rules](https://www.sarasoueidan.com/blog/horizontal-rules/#)

## Changelog

<ComponentChangelogTable changelogData={changelogData} />
