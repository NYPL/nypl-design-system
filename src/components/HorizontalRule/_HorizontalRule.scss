.horizontal-rule {
  background-color: $gray-light;
  border: 0;
  height: 2px;
  margin-bottom: var(--space-s);
  margin-top: var(--space-s);

  &--left {
    margin-left: 0;
  }

  &--right {
    margin-right: 0;
  }
}
