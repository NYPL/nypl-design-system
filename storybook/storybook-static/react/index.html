<!doctype html><html lang="en"><head><meta charset="utf-8"><title>Storybook</title><meta name="viewport" content="width=device-width,initial-scale=1"><style>html, body {
    overflow: hidden;
    height: 100%;
    width: 100%;
    margin: 0;
    padding: 0;
  }</style><script>/* globals window */
  /* eslint-disable no-underscore-dangle */
  try {
    if (window.top !== window) {
      window.__REACT_DEVTOOLS_GLOBAL_HOOK__ = window.top.__REACT_DEVTOOLS_GLOBAL_HOOK__;
    }
  } catch (e) {
    // eslint-disable-next-line no-console
    console.warn('unable to connect to top frame for connecting dev tools');
  }</script><style>#root[hidden],
      #docs-root[hidden] {
        display: none !important;
      }</style></head><body><div id="root"></div><div id="docs-root"></div><script>window['DOCS_MODE'] = false;</script><script src="runtime~main.00bd9d033d7a0b972fc9.bundle.js"></script><script src="vendors~main.d2b6aafef1a259d165c0.bundle.js"></script><script src="main.3bf71672a6dd35499004.bundle.js"></script></body></html>