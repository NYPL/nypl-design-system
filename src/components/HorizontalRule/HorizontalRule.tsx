// HorizontalRule
import * as React from "react";
import bem from "../../utils/bem";

export interface HorizontalRuleProps {
  /** Optional alignment value to align the horizontal rule to one side or the other when the width is less than 100%. If omitted, the horizobntal rule will have a default center alignment. */
  align?: "left" | "right";
  /** Additional attributes passed to the horizontal rule */
  attributes?: { [key: string]: any };
  /** ClassName you can add in addition to `horizontal-rule` */
  className?: string;
  /** Optional height value.  This value should be entered with the same formatting as a CSS height attribute (ex. `2%`, `5px`, `1.5rem`).  If omitted, the horizobntal rule will have a default height of 2px. */
  height?: string;
  /** Modifiers array for use with BEM. See how to work with modifiers and BEM here: http://getbem.com/introduction/ */
  modifiers?: string[];
  /** Optional width value.  This value should be entered with the same formatting as a CSS width attribute (ex. `50%`, `640px`, `20rem`).  If omitted, the horizobntal rule will have a default width of 100%. */
  width?: string;
}

export default function HorizontalRule(
  props: React.ComponentProps<"hr"> & HorizontalRuleProps
) {
  const {
    align,
    attributes = {},
    className,
    width = "auto",
    height = "2px",
    modifiers = [],
  } = props;

  attributes["style"] = {
    width: width,
    height: height,
  };

  if (align) {
    modifiers.push(align);
  }

  return (
    <hr
      className={bem("horizontal-rule", modifiers, "", [className])}
      {...attributes}
    />
  );
}
