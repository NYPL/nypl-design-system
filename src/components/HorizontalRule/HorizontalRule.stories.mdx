import {
  Meta,
  Story,
  Canvas,
  ArgsTable,
  Preview,
  Description,
} from "@storybook/addon-docs/blocks";
import { withDesign } from "storybook-addon-designs";

import HorizontalRule from "./HorizontalRule";
import { action } from "@storybook/addon-actions";

<Meta
  title="HorizontalRule"
  component={HorizontalRule}
  decorators={[withDesign]}
  parameters={{
    design: {
      type: "figma",
      url:
        "https://www.figma.com/file/qShodlfNCJHb8n03IFyApM/Main?node-id=19659%3A0",
    },
  }}
  argTypes={{
    attributes: { table: { disable: true } },
    modifiers: { table: { disable: true } },
  }}
/>

# Horizontal Rule

<Description of={HorizontalRule} />

The `HorizontalRule` component renders a basic `<hr>` element. The `height`, `width` and `alignment` can be customized.

<Preview withToolbar>
  <Story name="HorizontalRule">{args => <HorizontalRule {...args} />}</Story>
</Preview>

<ArgsTable story="HorizontalRule" />
