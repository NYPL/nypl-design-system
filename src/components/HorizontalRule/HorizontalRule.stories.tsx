import type { Meta, StoryObj } from "@storybook/react";
import { withDesign } from "storybook-addon-designs";

import HorizontalRule from "./HorizontalRule";

const meta: Meta<typeof HorizontalRule> = {
  title: "Components/Page Layout/HorizontalRule",
  component: HorizontalRule,
  decorators: [withDesign],
  argTypes: {
    className: { control: false },
  },
};

export default meta;
type Story = StoryObj<typeof HorizontalRule>;

/**
 * Main Story for the HorizontalRule component. This must contains the `args`
 *  and `parameters` properties in this object.
 */
export const WithControls: Story = {
  args: { align: undefined, className: undefined },
  parameters: {
    design: {
      type: "figma",
      url: "https://www.figma.com/file/qShodlfNCJHb8n03IFyApM/Main?node-id=36818-25303&t=R7gWIzMTzUAGN7Sw-4",
    },
    jest: "HorizontalRule.test.tsx",
  },
};
