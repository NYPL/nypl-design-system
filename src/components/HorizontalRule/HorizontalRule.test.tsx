import { expect } from "chai";
import * as Enzyme from "enzyme";
import * as React from "react";
import HorizontalRule from "./HorizontalRule";

describe("HorizontalRule: check for basic element", () => {
  let container;
  before(() => {
    container = Enzyme.mount(<HorizontalRule />);
  });

  it("Renders HorizontalRule component", () => {
    expect(container.find("hr").exists()).to.equal(true);
  });
});

describe("HorizontalRule: check for custom class", () => {
  let container;
  before(() => {
    container = Enzyme.mount(<HorizontalRule className={"custom-hr"} />);
  });

  it("Renders HorizontalRule component", () => {
    expect(container.find("hr").exists()).to.equal(true);
  });

  it("Renders with proper custom class", () => {
    expect(container.find("hr").hasClass("custom-hr")).to.equal(true);
  });
});

describe("HorizontalRule: check for custom dimensions", () => {
  let container;
  before(() => {
    container = Enzyme.mount(
      <HorizontalRule className={"custom-hr"} width={"720px"} height={"5rem"} />
    );
  });

  it("Renders HorizontalRule component", () => {
    expect(container.find("hr").exists()).to.equal(true);
  });

  it("Renders with correct width and height values", () => {
    expect(container.find("hr").get(0).props.style).to.have.property(
      "width",
      "720px"
    );
    expect(container.find("hr").get(0).props.style).to.have.property(
      "height",
      "5rem"
    );
  });
});

describe("HorizontalRule: check for custom width and custom alignment", () => {
  let container;
  before(() => {
    container = Enzyme.mount(
      <HorizontalRule className={"custom-hr"} width={"80%"} align={"left"} />
    );
  });

  it("Renders HorizontalRule component", () => {
    expect(container.find("hr").exists()).to.equal(true);
  });

  it("Renders with correct width and height values", () => {
    expect(container.find("hr").get(0).props.style).to.have.property(
      "width",
      "80%"
    );
    expect(container.find("hr").get(0).props.style).to.have.property(
      "height",
      "2px"
    );
  });

  it("Renders with correct alignment", () => {
    expect(container.find("hr").hasClass("horizontal-rule--left")).to.equal(
      true
    );
  });
});
