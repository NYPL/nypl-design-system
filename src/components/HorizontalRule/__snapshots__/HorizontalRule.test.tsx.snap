// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HorizontalRule renders the UI snapshot correctly 1`] = `
<hr
  className="css-ili6fu"
/>
`;

exports[`HorizontalRule renders the UI snapshot correctly 2`] = `
<hr
  className="css-1pwp4fu"
/>
`;

exports[`HorizontalRule renders the UI snapshot correctly 3`] = `
<hr
  className="css-gj8znx"
/>
`;

exports[`HorizontalRule renders the UI snapshot correctly 4`] = `
<hr
  className="css-ili6fu"
/>
`;

exports[`HorizontalRule renders the UI snapshot correctly 5`] = `
<hr
  className="css-ili6fu"
/>
`;
